projects/RabbitMQ.Client/client/impl/OutgoingCommand.cs

```diff
@@ -83,34 +109,14 @@ internal void Transmit(ushort channelNumber, Connection connection)
                 }
             }
 
+#if DEBUG
```

Comment:
Maybe we'd want to use `Debug.Assert(offset != size, $"Serialized to wrong size, expect {size}, offset {offset}");` instead of `#if DEBUG`

Reply:
Good idea

Reply:
The #if Debug can be removed now, as the Debug.Assert is only compiled for Debug anways.